Determinant appendix

Xintong-Li opened the pull request:
Appendix: determinant (draft)

This PR is about the determinant appendix.

Some Problems:
1. To illustrate the diagonal method, I used the `xymatrix` package. However, this package interally override `\Col` and `\Row`. To compensate for this, I included this package at the beginning of appendix 3 (resulting in a "can only be used in preample" error).

2. To illustrate the cofactor method, I used the `nicematrix` package. I included this package in the `preample.tex`

Update: I used `nicematrix` to recreate the diagonal trick part so that `xymatrix` is no longer needed.

siefkenj commented:
Run your code through https://siefkenj.github.io/latex-parser/   It'll make the formatting nice.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+Determinant is useful: it measures the volume distortion of a linear transformation, it helps us determine if a matrix is invertible of not, and it provides us with a simple formula for solving \(A\vec x=\vec y\) when \(\vec x\) is invertible.
```
`x` is invertible?

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+Determinant is useful: it measures the volume distortion of a linear transformation, it helps us determine if a matrix is invertible of not, and it provides us with a simple formula for solving \(A\vec x=\vec y\) when \(\vec x\) is invertible.
+
+In Module 14, you will learn an algorithm for evaluating determinants via row reduction. However, for \(2\times 2\) and \(3\times 3\) determinants, we have methods which are more convenient than row reduction. We will explore these methods in this appendix.
```
Add a `\label` to the start of Module 14 (right after the `\Title`) and then use a `\ref`. No absolute references allowed.

siefkenj replied:
Keep a consistent tense: "Module 14 discusses ..."

siefkenj replied:
They aren't more convenient. They are direct formulas.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+\Heading{A formula for \(2\times 2\) determinants}
+For a \(2\times 2\) matrix, once we know its four entries, we can calculate its determinant directly.
+\begin{theorem}[\(2\times 2\) determinant formula]
+  Let \(M\) be a \(2\times 2\) matrix. Suppose \(M=\mat{a & b \\ c & d}\). Then \[\det(M)=ad-bc.\]
```
Let `M=\mat{...}`. Then, ...

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+\begin{example}
+Use the \(2\times 2\) determinant formula to find \(\det(M)\), where \(M=\mat{1 & 6 \\ 2  & 7}\).
+\end{example}
+\textbf{Solution. }Apply the formula, and we get \[\det(M)=1\times 7-2\times 6=-5.\]
```
Solution goes inside the example. Don't prefix the solution with `Solution.`

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+Now we outline the procedure of calculating \(\det(\mat{a & b & c \\ d & e & f \\ g & h & i})\) via this trick.
+
+\textbf{Step 1. }Copy the matrix, and then copy the \emph{first two columns} to the right of the matrix:
+\[\begin{NiceArray}{CCCCCCCCCCC}[margin]
```
Why don't you do this with the augmented matrix syntax? Then you can use `{\color{...} ...}` in the cells to distinguish the columns/diagonals.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,172 @@
+
+Now we outline the procedure of calculating \(\det(\mat{a & b & c \\ d & e & f \\ g & h & i})\) via this trick.
+
+\textbf{Step 1. }Copy the matrix, and then copy the \emph{first two columns} to the right of the matrix:
```
Use an `\begin{itemize}` if you are going to list steps. You can then do `\item[Step 1]`

Xintong-Li commented:
Update:

1. Used Latex Parser to reformat the source code.

2. Fixed some wording and typographical problems pointed out by previous PR, as well as other typos.

Some problems (to be solved before merging):

1. I am now using `itemize` environment for diagonal trick. However, it results in the following problem:

![image](https://user-images.githubusercontent.com/35551659/86341273-137c9e00-bc24-11ea-91ca-2b566d6fe69a.png)

Is there a way to solve it?

2. To illustrate diagonal trick, now I use the augmented matrix syntax. Currently the effect is the as first matrix illustrated below.
Some comments:
(a) Would it be better if I use `vmatrix` instead of `bmatrix`?
(b) For the two hightlighting methods, which is appears to be better?
(I commented the code for the second highlighting methods. It need to be deleted before merging.)

![image](https://user-images.githubusercontent.com/35551659/86341103-d6b0a700-bc23-11ea-9ba1-22ece9832a0a.png)

siefkenj commented:
Look at the documentation for the `enumitem` package. It will tell you how to adjust the left margin of an `itemize` to give more space. Also, we should use an `enumerate` instead, since we're actually enumerating.

The background color highlighting looks great! We need to make sure it also works for color blind people, so pick colors that you can tell the difference between in a black-and-white screenshot.

Xintong-Li commented:
Update highlighting.

Now, the diagonal trick illustrations look like 
![image](https://user-images.githubusercontent.com/35551659/86399385-2ec3c980-bc75-11ea-83d3-c12ead100e68.png)
In black and white, they look like 
![image](https://user-images.githubusercontent.com/35551659/86400029-e527ae80-bc75-11ea-80ae-a7902e3f6d8c.png)

siefkenj commented:
Certainly looks nice in black and white. Can you find more pleasing colors, though?  These color scales work well in B&W: https://www.thinkingondata.com/something-about-viridis-library/

siefkenj commented on book/modules/appendix3-exercises.tex:
```diff
@@ -0,0 +1,96 @@
+				theorem from Module \ref{determinant module}.)
+
+			\item Inspired by your argument, Kokoro then developed an argument
+				about the invalidity of expanding along the third columns: \textit{If
```
Use `\emph`, not `\textit`

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,241 @@
+	Thus, $\det\mat{1 & 4 & 0 \\ -2 & 3 & 1 \\ 0 & 2 & 1}=3-(-6)=9$.
+\end{example}
+
+\Heading{Calculating $3\times 3$ determinants (II): Expanding along the first column}
```
Title Capitalization for Headings

luema commented:
> Update highlighting.
> 
> Now, the diagonal trick illustrations look like
> ![image](https://user-images.githubusercontent.com/35551659/86399385-2ec3c980-bc75-11ea-83d3-c12ead100e68.png)
> In black and white, they look like
> ![image](https://user-images.githubusercontent.com/35551659/86400057-f375ca80-bc75-11ea-94c6-b6610578e8ea.png)

Just to add to what Jason said about the colour, I've used this site in the past to find colourblind-friendly palettes: https://colorbrewer2.org/#type=qualitative&scheme=Dark2&n=3

luema commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,241 @@
+via row reduction. However, for $2\times 2$ and $3\times 3$ determinants, we
+have direct formulas. We will explore these formulas in this appendix.
+
+\Heading{A formula for $2\times 2$ determinants} For a $2\times 2$ matrix, once we
```
I think there should be some consistency among the headings. Something like `Computing 2x2 Determinants`, `Computing 3x3 Determinants (I)...`, `Computing 3x3 Determinants (II)...`, etc.

Xintong-Li commented:
Update: Color and other issues

1. Updated color. The color palette is inspired by _3-class Paired_ palette on https://colorbrewer2.org/#type=qualitative&scheme=Paired&n=3

![image](https://user-images.githubusercontent.com/35551659/86461830-966f2880-bcf8-11ea-9c00-3812df269ce5.png)
![image](https://user-images.githubusercontent.com/35551659/86461847-9cfda000-bcf8-11ea-96a5-4bd089cac4b8.png)

2. Reformatted the appearence of determinants. Now arugment of any determinant function always has a pair of parentheses (of suitaible size) surrounding it.

3. Reformatted the appearence of fractions. If a fraction is accompanied by a matrix, then the fraction is set to `displaystyle`.

4. Remove a problem about vector products (because there's already a problem reg. vector products in tutorial). Added new problems: determine $\operatorname{Vol~Change}(\mathcal{T})$ for a given transformation $\mathcal{T}:\mathbf{R}^3\rightarrow\mathbfR}^3$, and determine, using determinant, if $\operatorname{Col}(A)$ has the max possible dimension.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,241 @@
+\]
+Using the summation notation, this can be written in a more compact form:
+\[
+	\det(A)=\sum_{r=1}^{3}(-1)^{r+1}a_{r1}M_{r1}.\footnote{In fact, this can be generalized to determinants of arbitriary size: for any $n
```
`.` after footnote.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,241 @@
+
+\Heading{Calculating $3\times 3$ Determinants (II): Expanding along the First Column}
+The second method of calculating $3\times 3$ determinants actually uses $2\times
+2$ determinants. This method is called \textbf{expanding along the first column}.
```
`textbf` -> `emph`

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,241 @@
+Before moving on, we give an explanation about the name ``expanding along the first
+column''.
+
+The formula involves three determinants. If we delete the first column and first row of \(A\). we are left with a \(2\times 2\) matrix (below, a highlighted column/row means that this column/row is deleted).
```
Run through `latex-parser`

Xintong-Li commented:
Update: 

1. Modified exercise 3. Now it is a question related to basis (including orientation) and determinant.
2. Modified exercise 6. Now it is a T/F question.
3. Added a warning box.
4. Added subentries of `Determinant` in index: `$2\times 2$ formula`, `$3\times 3$ formula`, `$3\times 3$ diagonal trick`, and `$3\times 3$ cofactor expansion`.

Comment:
Warning box is not something predefined. Thus, I created a warning box using `emphbox`. The problem is, this warning box appears in green, which seems to not be a warning. Perhaps I need to find a way to temparily change the color of this emphbox.

TODO:
Add a geometrical proof of the 2-by-2 formula.

Xintong-Li commented:
**Update:** Added a geometrical proof of the 2-by-2 formula.

**Comment. ** I am trying to give a proof where no square roots or triangle ratios are involved. However, I met the problem that I cannot find a "unified" way of proving the theorem.  Thus, I seperated the proof into several cases.

In addition, I am not quite familiar with the package `Tikz`, so I just modified the diagram in another section.

siefkenj commented on book/linearalgebra.tex:
```diff
@@ -8925,6 +8924,19 @@ \section*{Eigenvectors}
 \end{appendix}
 
+\begin{appendix}
+	\Title{Determinants of Order 2 and 3}
```
-> `Formulas for 2x2 and 3x3 Determinants`

siefkenj commented on book/linearalgebra.tex:
```diff
@@ -8925,6 +8924,19 @@ \section*{Eigenvectors}
+	\Title{Determinants of Order 2 and 3}
+	In this appendix you will learn:
+\begin{itemize}
+  \item A practical formula for \(2\times 2\) determinants.
```
`\(..\)` -> `$...$`

siefkenj commented on book/modules/appendix3-exercises.tex:
```diff
@@ -0,0 +1,74 @@
+\begin{exercises}
+	\begin{problist}
+		\prob For each matrix given below, calculate its determinant by (1) row reduction, and (2) using the $2\times 2$ determinant formula.
```
Format with the online formatter.

siefkenj commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,476 @@
+	\]
+\end{theorem}
+\begin{proof}
+Let $\mathcal{T}_{M}$ be the matrix multiplication mapping induced by \(M\), namely \[\mathcal{T}_{M}:\mathbb{R}^2\rightarrow \mathbb{R}^2,\quad \text{ where }\mathcal{T}(\vec x)=M\vec x.\]
```
Format with the online formatter.

siefkenj commented:
Any figure you're not sure how to do, you can mark with `XXX Figure` and a description of the figure and we can fill it in later.

JackC92 commented on book/modules/appendix3.tex:
```diff
@@ -0,0 +1,476 @@
+\begin{proof}
+Let $\mathcal{T}_{M}$ be the matrix multiplication mapping induced by \(M\), namely \[\mathcal{T}_{M}:\mathbb{R}^2\rightarrow \mathbb{R}^2,\quad \text{ where }\mathcal{T}(\vec x)=M\vec x.\]
+
+If $\Set{\matc{a \\ c},\matc{b \\ d}}$ is linearly dependent, then $\dim \Range(\mathcal{T}_M)<2$, then $\mathcal {T}_{{M}}(C_2)$ is either a point or a one-dimensional object. In either case it has signed area $0$, so $\det(\mathcal{T}_{M})=0$. Thus, $\det(M)=0$. Now, the fact that $\Set{\matc{a \\ c},\matc{b \\ d}}$ is linearly dependent implies that there exists a scalar $\lambda$ such that $\ ...
```
You can use `\Set*{}` for the enlarged curly brackets.

siefkenj replied:
`C_2` is the unit 2-cube. But it would be good to remind the reader of that.

Xintong-Li commented:
Update: Formatting

Formatted the proof using LaTeX Parser.

Change all manually crafted set braces to `\Set*{}`
